src/Lightning.ts

```diff
@@ -720,17 +715,31 @@ export const LightningMixin = (superclass) =>
             result = await getPayment({ lnd, id: payment.hash })
           } catch (err) {
             const error = "issue fetching payment"
-            this.logger.error({ lnd, err, payment }, error)
-            throw new LoggedError(error)
+            paymentLogger.error({ lnd, err }, error)
+            continue
           }
 
           if (result.is_confirmed || result.is_failed) {
-            payment.pending = false
-            await payment.save()
+            const resultPendingTrue = await Transaction.updateMany(
+              { _journal: payment._journal, pending: true },
+              { $set: { pending: false } },
+            )
+
+            if (resultPendingTrue.nModified === 0) {
+              // this could happen if dealer and user try to update transaction at the same time
```

Comment:
Does this mean there was a race condition between the dealer code and the 'main' code? Was it because the dealer transactions weren't lock protected?

Reply:
what was not protected is if both the dealer (which is a user) and the user were both trying to update the transaction at the same time.

but the root issue here that we have is that we were doing a `updateOne` (in the form of `payment.save()`) instead of `updateMany`. so we were only updating some entries for the transaction.

Reply:
@bodymindarts when we add a payment to the user we create an entry in medici that basically creates at least 2 Transactions (credit and debit) with the same metadata but for different accounts.

So the problem was fixed by `updateMany` because when we update the transactions owned by the user we also update the transaction for lightning assets account (lndAccountingPath)

Also, we went through the code to check if it was happening elsewhere, but everything is fine.

With the refactoring in the first stage of [usd invoices issue](https://github.com/GaloyMoney/galoy/issues/350#issuecomment-885293504) we will be able to tackle these kinds of things easily